sensor/admission-control/alerts/sender_test.go

```diff
@@ -0,0 +1,158 @@
+package alerts
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+)
+
+func TestAlertSender(t *testing.T) {
+	suite.Run(t, new(alertSenderSuite))
+}
+
+type alertSenderSuite struct {
+	suite.Suite
+	service *alertSenderImpl
+	cl      *fakeAdmissionControlManagementServiceClient
+}
+
+func (s *alertSenderSuite) TestSendAlertsToSensor() {
+	ctx := context.Background()
+	ch := make(chan []*storage.Alert)
```

Comment:
Wouldn't it be less flaky to create this channel inside the test `Run` callback?

If there's a regression and one of the test cases fail early without reading from the queue this could fail all other tests, right?

Reply:
Good catch. I’ve moved the creation of the channel inside `Run`.